src/php/Compiler/Domain/Parser/ExpressionParser/AtomParser.php

```diff
@@ -60,11 +60,7 @@ public function parse(Token $token): AbstractAtomNode
             return $this->parseOctalNumber($matches, $word, $token);
         }
 
-        if (is_numeric($word)) {
-            return new NumberNode($word, $token->getStartLocation(), $token->getEndLocation(), $word + 0);
-        }
-
-        if (preg_match(self::REGEX_DECIMAL_NUMBER, $word, $matches)) {
+        if (is_numeric($word) || preg_match(self::REGEX_DECIMAL_NUMBER, $word)) {
```

Comment:
This change broke the tests, @JesusValera 😆

Reply:
I am trying to understand why the test is failing and it seems the part of
```php
$parser->parse(
    new Token(Token::T_ATOM, '1.2', $start, $end),
),
```
in the test is not creating a "proper" `NumberNode` as the `value` is 1, instead of `1.2`, is that an error @jenshaase?


![image](https://user-images.githubusercontent.com/6381924/216446298-70133051-5ea3-4804-9bfb-f16950ef4c31.png)

Reply:
But... the test shouldn’t fail, no? Maybe @jenshaase can help us understanding if the test is valid, or wrong and we need to update it 🤔

Reply:
The problem is in the following method
```php
private function parseDecimalNumber(array $matches, string $word, Token $token): NumberNode
{
    $sign = (isset($matches[1]) && $matches[1] === '-') ? -1 : 1;

    return new NumberNode(
        $word,
        $token->getStartLocation(),
        $token->getEndLocation(),
        $sign * (int)str_replace('_', '', $word),
    );
}
```
In the last line, we are casting `$word` to integer, so, for decimal numbers like `1.2` are transformed into `1`. That's why some tests fail. I also tried to change the cast to `float` as follows
```php
$sign * (float)str_replace('_', '', $word),
```
but then, there are a lot of errors everywhere.

Changes were rollback, like this is ok 👌🏼